third_party/k3d/k3d.go

```diff
@@ -251,8 +253,25 @@ func (c *Cluster) GetKubectlContext() string {
 }
 
 func (c *Cluster) ExportLogs(ctx context.Context, dest string) error {
-	log.Warning("ExportLogs not implemented for k3d. Please use regular kubectl like commands to extract the logs from the cluster")
-	return nil
+	reqVersion, err := semver.Parse(c.version)
+	supVersion, _ := semver.Parse(k3dWithExportLogSupport)
+	if err != nil {
+		log.ErrorS(err, "failed to determine the k3d version to decide if the current version supporst the log export helpers. Please use regular kubectl like commands to extract the logs from the cluster")
+		return nil
+	}
+	var stdout, stderr bytes.Buffer
+	if reqVersion.GE(supVersion) {
+		p := utils.RunCommandWithCustomWriter(fmt.Sprintf("%s debug export-logs %s --path %s", c.path, c.name, dest), &stdout, &stderr)
+		err = p.Err()
+		if err != nil {
+			log.ErrorS(err, "failed to export cluster logs due to an error", "stdout", stdout.String(), "stderr", stderr.String(), "result", p.Result())
+			return err
+		}
+		return nil
+	} else {
+		log.Warning("ExportLogs not implemented for k3d. Please use regular kubectl like commands to extract the logs from the cluster")
```

Comment:
Do we know what that `kubectl` command would be to extract logs? Could we just use it as a fallback instead of just reporting a warning?

Reply:
@vladimirvivien  Actually, That `kubectl like` is probably a mistake. Just `kubectl` won't be enough. We will also have to get a few more logs. And the problem is, we have to find a way to group them in a proper format. it can be tricky. (Also a duplicate work to replicate the logic).  

If we want to add an export mechanism, we might be better off writing some mechanism that can do this for any provider that doesn't have a log export support. 

`vcluster` doesn't support. So is the case with `kwok`. 

For `kwok` there is an adhoc `kubectl logs` equivalent of export but it does only for the `kwok`'s own components. 

How do we go about doing this ?

Reply:
We also have to worry about fetching the journalctl logs + the dmesg + any other applicable logs. Might not be a bad idea to implement something generic that can be reused for any provider in the future that is lacking the log export capabilities

Reply:
@vladimirviven ptal.. I would love tp get this one merged.